provider/internal/java/dependency.go

```diff
@@ -39,14 +40,25 @@ func (p *javaServiceClient) findPom() string {
 }
 
 func (p *javaServiceClient) GetDependencies() (map[uri.URI][]*provider.Dep, error) {
-	ll, err := p.GetDependenciesDAG()
-	if err != nil {
-		return p.GetDependencyFallback()
-	}
-	if len(ll) == 0 {
-		return p.GetDependencyFallback()
-	}
+	var err error
+	var ll map[uri.URI][]konveyor.DepDAGItem
 	m := map[uri.URI][]*provider.Dep{}
+	if p.isLocationBinary {
+		ll = make(map[uri.URI][]konveyor.DepDAGItem, 0)
+		// for binaries we only find JARs embedded in archive
+		w := walker{
+			deps: ll,
+		}
+		filepath.WalkDir(p.config.DependencyPath, w.walkDirForJar)
```

Comment:
I think we use this code somewhere else too right? I wonder if it could always be used here only and then added to out of the others if not binary so we only do this code in one place?